Actions/RunPipeline/RunPipeline.ps1

```diff
@@ -119,6 +124,100 @@ try {
         exit
     }
 
+    $buildArtifactFolder = Join-Path $projectPath ".buildartifacts"
+    New-Item $buildArtifactFolder -ItemType Directory | Out-Null
+
+    if ($baselineWorkflowSHA -and $baselineWorkflowRunId -ne '0' -and $settings.incrementalBuilds.mode -eq 'modifiedApps') {
+        # Incremental builds are enabled and we are only building modified apps
+        $headSHA = git rev-parse HEAD
+        Write-Host "Current HEAD is $headSHA"
+        git fetch origin $baselineWorkflowSHA | Out-Host
+        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to fetch baseline SHA $baselineSHA" }
```

Comment:
Not sure what a user would expect, but can we fall back to just building all apps in the project? Would be annoying to see your build fail on a git fetch issue IMO

Reply:
.. or could there be retries?

Reply:
Will rebuild all if git fetch for some reason fails

Reply:
I recall `git fetch` fails occasionally. My opinion is that retries is the better approach here.

This is what we did in BCApps, for example: https://github.com/microsoft/BCApps/commit/9b6cf1d23b2cabd88a955597928093fedbe5526d

Reply:
Read about reasons for why git fetch can fail and you are right, there are legitimate reasons why a retry would help git fetch - have adopted the functionality from BCApps.
Looking at this - we should create functions like this in a PowerShell module, which BCApps could get from the version of AL-Go it is using - like we are discussing with other BcContainerHelper functions and probably other helper functions, which are in both repos - I don't think this refactoring should be part of this PR though.